plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/gitlab.ts

```diff
@@ -149,19 +157,29 @@ export function createPublishGitlabAction(options: {
         id: number;
       };
 
+      const { id: userId } = (await client.Users.current()) as {
+        id: number;
+      };
+
       if (!targetNamespace) {
-        const { id } = (await client.Users.current()) as {
-          id: number;
-        };
-        targetNamespace = id;
+        targetNamespace = userId;
       }
 
-      const { http_url_to_repo } = await client.Projects.create({
+      const { id: projectId, http_url_to_repo } = await client.Projects.create({
         namespace_id: targetNamespace,
         name: repo,
         visibility: repoVisibility,
       });
 
+      if (setUserAsOwner && integrationConfig.config.token) {
+        const adminClient = new Gitlab({
```

Comment:
I'm not sure that we should be creating a new client here. We already have a token that we used further up, and that token was used to create the repo, so it should have the required scopes to update the owner.  I recommend we use the `client` already defined to further up.

Reply:
Probably don't need to check if there's an `integrationConfig.config.token` then either.

Reply:
This workflow is supposed to work when the OAuth users do not have the rights to set themselves as owners of the repo. That's why I need to use the token in the integration config, because the users in our case were not able to push the initial protected branch nor setting themselves as owners.

Reply:
To summarize:
- the user can create the repo and that's it. It cannot push on it. I know it's weird, but it's how gitlab works...
- only thing i could try to do is see, if I can use a different token for a single request, without creating a new client

Reply:
> the users can create the repo and that's it. They cannot push on it

this makes absolutely zero sense :joy: wasn't aware it worked like this, but ok, now I see the issue. 

Ok, let's go with this then, even it's a little unfortunate.